eclipse-sw360/sw360
# feat(rest):Update API Create Release with Cots Detail

[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)
> * Did you add or update any new dependencies that are required for your change?


### How To Test?

- Run command line: 
```
curl 'https://sw360.org/api/releases' -i -X POST \
    -H 'Content-Type: application/hal+json' \
    -H 'Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOlsic3czNjAtUkVTVC1BUEkiXSwidXNlcl9uYW1lIjoiYWRtaW5Ac3czNjAub3JnIiwic2NvcGUiOlsiYWxsIl0sImV4cCI6MTY5MjAwNDYyOSwiYXV0aG9yaXRpZXMiOlsiUkVBRCIsIldSSVRFIl0sImp0aSI6IkxoS1l1aEItbUJSeVV1QWNLT2FseXRSWW1yYyIsImNsaWVudF9pZCI6InRydXN0ZWQtc3czNjAtY2xpZW50In0.pObj6HM4dKGeEK0weAX8TLGt3Da9nrkXxFwbjdLsEylnRID1gDax_NonMYvHAI8gEqRzZtE7FJ2CAhveJiSIH8rwzDh8jzbJL-o0G3NRZylmYBtv4uAe8rbSkNrLuzRrjgtorQyl7AlUdwlAd4VoCVAeJ5iB1-kIgrrMrsFk-6K3yOtxSCF3IlMIbq8NuR-UthkiwM1DuQaBIF-Zcf5Dv8rJU-MQkAuKeqrrXxpo_SHsJqdBa4t59oTH5g_GtFEvKJw2Ww3dTzbmBf844JSmVcNx6eH7CV1DdhL6DcGXcMk_d4Bfz0vPKaHHunoIUvBgZ76RZM7SCYyP5q5PS2hrYw' \
    -d '{
    "name": "Spring Core",
    "version": "1.0",
    "componentId": "e1fe38c315494387bcb64319024bcde1",
    "cotsDetails": {
        "usedLicense": "",
        "licenseClearingReportURL": "https://www.google.com/search",
        "containsOSS": false,
        "ossContractSigned": true,
        "ossInformationURL": "https://www.google.com/search",
        "usageRightAvailable": true,
        "cotsResponsible": "user@sw360.org",
        "clearingDeadline": "2023-08-31",
        "sourceCodeAvailable": true
    }
}'

```

- Expected Output

```
{
    "name": "Spring Core",
    "version": "1.0",
    "cotsDetails": {
        "usedLicense": "",
        "licenseClearingReportURL": "https://www.google.com/search",
        "containsOSS": false,
        "ossContractSigned": true,
        "ossInformationURL": "https://www.google.com/search",
        "usageRightAvailable": true,
        "cotsResponsible": "user@sw360.org",
        "clearingDeadline": "2023-08-31",
        "sourceCodeAvailable": true
    }
    "_links": {
        "sw360:component": {
            "href": "http://localhost:8080/resource/api/components/e1fe38c315494387bcb64319024bcde1"
        },
        "self": {
            "href": "http://localhost:8080/resource/api/releases/6bed1b93ebc54c8498a45f63d4e264b9"
        },
        "curies": [
            {
                "href": "http://localhost:8080/resource/docs/{rel}.html",
                "name": "sw360",
                "templated": true
            }
        ]
    }
}
```
